### ISSUE TITLE
Purge deleted messages prompt missing

### ISSUE THREAD

**Issue by l2dy** [neomutt/neomutt#2028]
## Expected Behaviour
I get a prompt of `Purge x deleted messages? ([yes]/no):` on exit.

## Actual Behaviour
No prompt on exit.

## Steps to Reproduce
1. Delete a message to trash
2. Open the trash mailbox
3. Delete any mail in the trash
4. Quit NeoMutt

## How often does this happen?

- Always

## When did it start to happen?
20180716 was fine, still not fixed after 5bf1bac675d485c2b63d1408745bf5b3e70f73ea.

## NeoMutt Version

```
NeoMutt 20191207
Copyright (C) 1996-2016 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Darwin 18.7.0 (x86_64)
ncurses: ncurses 6.1.20180127 (compiled with 6.1.20180127)
libiconv: 1.16
libidn: 1.35 (compiled with 1.35)
GPGme: 1.13.1-unknown
hcache backends: gdbm

Configure options: --prefix=/opt/local --with-idn=/opt/local --with-ncurses=/opt/local --with-nls=/opt/local --with-ssl=/opt/local --with-gdbm=/opt/local --with-gpgme=/opt/local

Compilation CFLAGS: -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -arch x86_64 -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/opt/local/include -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash

Compile options:
  -autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens
  +getaddrinfo -gnutls +gpgme -gss +hcache -homespool +idn -inotify
  -locales_hack -lua +meta -mixmaster +nls -notmuch +openssl +pgp -sasl +smime
  -sqlite +start_color +sun_attachment +typeahead
MAILPATH="/var/mail"
PKGDATADIR="/opt/local/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/opt/local/etc"
```

## Extra Info

* Were you using multiple copies of NeoMutt at once?
No.
* Were you using 'screen' or 'tmux'?
No.
* Is your email local (maildir) or remote (IMAP)?
Local.

**Comment by gahr** [comment]
I get the "Purge x deleted messages" question when I switch from the first mailbox to the trash mailbox (or to any other mailbox, including quitting NeoMutt).

Please double check whether you get the prompt when doing the following:
1. start NeoMutt
2. `:exec delete-message` (please do not use any key bindings)
3. `:exec quit`

If you don't, please double check your `delete` setting.

**Comment by l2dy** [comment]
> I get the "Purge x deleted messages" question when I switch from the first mailbox to the trash mailbox (or to any other mailbox, including quitting NeoMutt).

Yes, but after switching to the trash mailbox and deleting any mail there and then quit. It would just quit without the prompt.

**Comment by gahr** [comment]
Do you have any folder-hook that changes the `delete` setting when you switch to the Trash folder? I consistently get the prompt in all folders. Can you try to run with a minimal config file?

**Comment by l2dy** [comment]
> Do you have any folder-hook that changes the `delete` setting when you switch to the Trash folder?

Nope.

> I consistently get the prompt in all folders. Can you try to run with a minimal config file?

I'll try that later.

**Comment by l2dy** [comment]
Fixed before commit eb047f6da97d85db8f4a93555fa1e420fea2c5b0. I'll try to bisect.

**Comment by l2dy** [comment]
896e08ec1d590c49f830e5699096dc79683e3be7 is the first bad commit.

Fixed in 815627e19ee760d71a0eb02dd404219ac3a70626.
